lib/datadog/core/telemetry/collector.rb

```diff
@@ -43,11 +43,20 @@ def application
 
         # Forms a hash of standard key value pairs to be sent in the app-started event configuration
         def configurations
+          peer_service_enabled = Datadog.configuration.tracing.contrib.global_default_service_name.enabled
+
           hash = {
             DD_AGENT_HOST: Datadog.configuration.agent.host,
             DD_AGENT_TRANSPORT: agent_transport,
             DD_TRACE_SAMPLE_RATE: format_configuration_value(Datadog.configuration.tracing.sampling.default_rate),
+            DD_TRACE_REMOVE_INTEGRATION_SERVICE_NAMES_ENABLED: peer_service_enabled
           }
+          peer_service_mapping_str = ''
+          unless Datadog.configuration.tracing.contrib.peer_service_mapping.empty?
+            peer_service_mapping = Datadog.configuration.tracing.contrib.peer_service_mapping
```

Comment:
Can you extract a local variable for `Datadog.configuration`, so this method becomes more succinct?
```suggestion
          config = Datadog.configuration

          hash = {
            DD_AGENT_HOST: Datadog.configuration.agent.host,
            DD_AGENT_TRANSPORT: agent_transport,
            DD_TRACE_SAMPLE_RATE: format_configuration_value(Datadog.configuration.tracing.sampling.default_rate),
            DD_TRACE_REMOVE_INTEGRATION_SERVICE_NAMES_ENABLED: config.tracing.contrib.global_default_service_name.enabled
          }
          peer_service_mapping_str = ''
          unless config.tracing.contrib.peer_service_mapping.empty?
            peer_service_mapping = config.tracing.contrib.peer_service_mapping
```